### PR TITLE
[version list] add method to get sonic version list

### PR DESCRIPTION
**Pull request by yxieca**
Summary:
Fixes # (issue)

### Type of change

- [] Bug fix
- [x] Testbed and Framework(new/improvement)
- [] Test case(new/improvement)

#### How did you do it?

add method to get sonic version list
Signed-off-by: Ying Xie <ying.xie@microsoft.com>

#### How did you verify/test it?
Return information would look like following:
{'current': u'SONiC-OS-20181130.74', 'installed_list': [u'SONiC-OS-20181130.51', u'SONiC-OS-wb-20200214.02', u'SONiC-OS-wb-20200214.01', u'SONiC-OS-20191130.26', u'SONiC-OS-20181130.74'], 'next': u'SONiC-OS-20181130.74'}

### CODE REVIEW

File: tests/common/devices.py [hunk 1]
```diff
@@ -312,6 +312,28 @@ def get_networking_uptime(self):
...
+    def get_image_info(self):
+        """
+        @summary: get list of images installed on the dut.
+                  return a dictionary of "current, next, installed_list"
+        """
+        output = self.command("sonic_installer list")["stdout"]
+        lines  = output.split('\n')
+        ret    = {}
+        list   = []
+        for line in lines:
```
Comment (daall) [inline tests/common/devices.py:324]: Python has a `splitlines()` method for this, so you can do:
```suggestion
        for line in output.splitlines():
```

File: tests/common/devices.py [hunk 2]
```diff
@@ -312,6 +312,28 @@ def get_networking_uptime(self):
...
 
+    def get_image_info(self):
+        """
+        @summary: get list of images installed on the dut.
+                  return a dictionary of "current, next, installed_list"
+        """
+        output = self.command("sonic_installer list")["stdout"]
+        lines  = output.split('\n')
+        ret    = {}
+        list   = []
```
Comment (daall) [inline tests/common/devices.py:323]: `list` is a keyword - maybe `image_list`?

Comment (tahmed-dev) [reply tests/common/devices.py:323]: In fact, there is another field that has stdout as lines, it is:

lines = self.command("sonic_installer list")["stdout_lines"]

Comment (tahmed-dev) [reply tests/common/devices.py:323]: the return value is documented in shell command:

https://docs.ansible.com/ansible/latest/modules/shell_module.html#return-values

File: tests/common/devices.py [hunk 3]
```diff
@@ -312,6 +312,28 @@ def get_networking_uptime(self):
...
+        for line in lines:
+            words = line.strip().split(' ')
+            if len(words) == 2:
+                if words[0] == 'Current:':
+                    ret['current'] = words[1]
+                elif words[0] == 'Next:':
+                    ret['next'] = words[1]
```
Comment (neethajohn) [inline tests/common/devices.py:330]: can use ret[words[0][:-1]] = words[1]

File: tests/common/devices.py (hunk 1 above)
Comment (yxieca) [reply tests/common/devices.py:324]: thanks. done!

File: tests/common/devices.py (hunk 2 above)
Comment (yxieca) [reply tests/common/devices.py]: thanks. changed!

File: tests/common/devices.py (hunk 3 above)
Comment (yxieca) [reply tests/common/devices.py:330]: Not sure I want to keep the first letter capital.
And also trying to keep the list specific due to some platform has extra output in certain versions.

File: tests/common/devices.py [hunk 4]
```diff
@@ -318,9 +318,9 @@ def get_image_info(self):
                   return a dictionary of "current, next, installed_list"
         """
         output = self.command("sonic_installer list")["stdout"]
-        lines  = output.split('\n')
+        lines  = output.splitlines()
```
Comment (neethajohn) [inline tests/common/devices.py:321]: not needed any more

Comment (jleveque) [reply tests/common/devices.py:321]: Why is this not needed? I believe this was the change that Danny requested below.

Comment (yxieca) [reply tests/common/devices.py:321]: oh, if we take Tamer's suggestion to use stdout_lines

Comment (neethajohn) [reply tests/common/devices.py:321]: My bad. I thought the var 'lines' wasn't used anymore. I was looking at Danny's suggested changes 'for line in output.splitlines()'

File: tests/common/devices.py (hunk 2 above)
Comment (yxieca) [reply tests/common/devices.py]: Changed according to Tamer's suggestions.

**Pull request merged**